src/index.tsx

```diff
@@ -1,6 +1,11 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import {HashRouter} from 'react-router-dom';
 import './index.css';
 import App from './App';
 
-ReactDOM.render(<App />, document.getElementById('root') as HTMLElement);
+ReactDOM.render((
+  <HashRouter>
```

Comment:
Why should we use `HashRouter`? (compared to `BrowserRouter`)

Reply:
I'm honestly not sure. I've read the docs, as well as a number of blog posts, and none of them seem to accurately explain the difference. In one place I saw someone mention that if you're running an SPA which does not require communication with a server for its functionality, then `HashRouter` makes it simpler. From everything I've read, other than that one mention, it doesn't seem like either would make much of a difference.

Also, I believe when we convert to `connected-react-router` it becomes moot, as there are no longer two different routers to choose between.

Reply:
So the `HashRouter` uses URLs like `/#foo/1`, which is convenient because you don't need to setup anything. The "server" serves `/` only and there is no need to hack anything to redirect all URLs to `/`. But URLs are fake from an HTTP point of view because everything after `#` is hidden from anything but the browser.

The `BrowserRouter` uses the HTML5 API, which is better because it uses proper URLs (real URLs), but it requires the server (that hosts the SPA) to redirect all URLs to `/`. This is pretty common use case nowadays, so it not a problem usually. It just works. I believe we should be using this one.

Reply:
Ok, I'll change it to `BrowserRouter`.